Support for Alpine Linux

The full build log can be found on: https://github.com/seia-soto/embeddable-wg/actions/runs/4303293368/jobs/7502847766

```
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@1.0.10
node-pre-gyp info using node@18.14.2 | linux | x64
node-pre-gyp info build requesting source compile
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.1
gyp info using node@18.14.2 | linux | x64
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.1
gyp info using node@18.14.2 | linux | x64
gyp info find Python using Python version 3.10.10 found at "/usr/bin/python3"
gyp http GET https://unofficial-builds.nodejs.org/download/release/v18.14.2/node-v18.14.2-headers.tar.gz
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v18.14.2/node-v18.14.2-headers.tar.gz
gyp http GET https://unofficial-builds.nodejs.org/download/release/v18.14.2/SHASUMS256.txt
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v18.14.2/SHASUMS256.txt
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/__w/embeddable-wg/embeddable-wg/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/github/home/.cache/node-gyp/18.14.2/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/github/home/.cache/node-gyp/18.14.2',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/github/home/.cache/node-gyp/18.14.2/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/__w/embeddable-wg/embeddable-wg',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@9.3.1
gyp info using node@18.14.2 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/__w/embeddable-wg/embeddable-wg/build'
  CC(target) Release/obj.target/wgbinding/adaptor/EmbeddableWireguardExtension.o
  CC(target) Release/obj.target/wgbinding/adaptor/napi_utils.o
  CC(target) Release/obj.target/wgbinding/externs/wireguard-tools/contrib/embeddable-wg-library/wireguard.o
../externs/wireguard-tools/contrib/embeddable-wg-library/wireguard.c:10:10: fatal error: linux/genetlink.h: No such file or directory
   10 | #include <linux/genetlink.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [wgbinding.target.mk:117: Release/obj.target/wgbinding/externs/wireguard-tools/contrib/embeddable-wg-library/wireguard.o] Error 1
make: Leaving directory '/__w/embeddable-wg/embeddable-wg/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/lib/build.js:203:23)
gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
gyp ERR! System Linux 5.15.0-1033-azure
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3/wgbinding.node" "--module_name=wgbinding" "--module_path=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=3" "--node_napi_label=napi-v3"
gyp ERR! cwd /__w/embeddable-wg/embeddable-wg
gyp ERR! node -v v18.14.2
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3/wgbinding.node --module_name=wgbinding --module_path=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/__w/embeddable-wg/embeddable-wg/node_modules/.pnpm/@mapbox+node-pre-gyp@1.0.10/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
node-pre-gyp ERR! System Linux 5.15.0-1033-azure
node-pre-gyp ERR! command "/usr/local/bin/node" "/__w/embeddable-wg/embeddable-wg/node_modules/@mapbox/node-pre-gyp/bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /__w/embeddable-wg/embeddable-wg
node-pre-gyp ERR! node -v v18.14.2
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/pnpm/dist/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3/wgbinding.node --module_name=wgbinding --module_path=/__w/embeddable-wg/embeddable-wg/lib/binding/napi-v3 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)
node-pre-gyp ERR! node-pre-gyp -v v1.0.10
node-pre-gyp ERR! not ok 
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -58,6 +58,30 @@ jobs:
         pattern: build/stage/**/wgbinding*.tar.gz
         github-token: ${{secrets.GITHUB_TOKEN}}
         release-url: ${{github.event.release.upload_url}}
+  build-alpine:
+    name: Build on Alpine Linux
+    runs-on: ubuntu-latest
+    container:
+      image: node:18-alpine
+    steps:
+      - name: Install build dependencies
+        run: apk update && apk add make g++ python3 git linux-headers
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Install package manager
+        run: npm i -g pnpm
+      - name: Install dependencies
+        run: pnpm install
+      - name: Package artifacts
+        run: pnpm build && pnpm node-pre-gyp package
+      - name: Upload to Release
+        uses: csexton/release-asset-action@v2
+        with:
+          pattern: build/stage/**/wgbinding*.tar.gz
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          release-url: ${{github.event.release.upload_url}}
   build-arm:
     name: Build on arm64
     runs-on: ubuntu-latest
@@ -66,6 +90,8 @@ jobs:
         include:
           - arch: aarch64
             distro: ubuntu20.04
+          - arch: aarch64
+            distro: alpine_latest
     steps:
     - name: Checkout
       uses: actions/checkout@v3
@@ -91,8 +117,8 @@ jobs:
               npm i -g pnpm
               ;;
             alpine*)
-              apk add --update make g++ python3
-              apk add --no-cache nodejs~=18 npm~=18
+              apk add --update make g++ python3 linux-headers
+              apk add --no-cache nodejs~=18 npm~=9
               npm i -g pnpm
               ;;
           esac

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "embeddable-wg",
-  "version": "0.0.1-d",
+  "version": "0.0.1-e",
   "description": "This package includes bindings of embeddable-wg-library in wireguard-tools library for efficient calls to set up WireGuard devices.",
   "main": "out/index.js",
   "type": "module",

